File: Programacion_1ºAño/T6Herencia/src/Trabajadores/model/Autonomos.java
package Trabajadores.model;

public abstract class Autonomos extends Trabajador implements Sindicador,Beneficio{
    private int cuotaSS;

    public Autonomos() {
    }


    public Autonomos(String nombre , String apellido , int nSsocial , double salario , int cuotaSS) {
        super(nombre , apellido , nSsocial , salario);
        this.cuotaSS = cuotaSS;
    }

    @Override
    public void calcularSalarioMes() {
        double salarioMes= getSalario()-(cuotaSS*12);
        System.out.println("El salario mensual al mes es de " +salarioMes);
    }

    @Override
    public void mostrarDatos() {
        super.mostrarDatos();
        System.out.println("Cuotas de la Seguridad Social = " +cuotaSS);
    }

    public int getCuotaSS() {
        return cuotaSS;
    }

    public void setCuotaSS(int cuotaSS) {
        this.cuotaSS = cuotaSS;
    }
}

File: Kotlin/TrabajadoresHerencia/src/model/Autonomos.kt
package model

class Autonomos(nombre: String, apellido: String, dni: String, salario: Double, var cuotasSS: Int, var seguro: Boolean) :
    Trabajador(nombre, apellido, dni, salario) {

    override fun calcularSalario(): Double {
        return salario - (cuotasSS * 12)
    }

    fun pedirDescuento(precioDes: Double) {
        if (seguro) {
            cuotasSS = (cuotasSS - cuotasSS * precioDes).toInt()
        } else {
            println("No ha sido posible aplicar descuento porque no hay seguro")
        }
    }

    override fun mostrarDatos() {
        super.mostrarDatos()
        println("Cuotas SS: $cuotasSS")
    }
}
